pkg/didcomm/transport/http/tls/certpool.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package tls
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"log"
+	"sync"
+	"sync/atomic"
+)
+
+// certPool is a thread safe wrapper around the x509 standard library
+// cert pool implementation.
+// It optionally allows loading the system trust store.
+type certPool struct {
+	certPool       *x509.CertPool
+	certs          []*x509.Certificate
+	certsByName    map[string][]int
+	lock           sync.RWMutex
+	dirty          int32
+	systemCertPool bool
```

Comment:
@Baha-sk this attribute does not seem to be required at all

Reply:
this attribute designates what its name implies, it's to use the system cert pool as opposed to using new and empty cert pool. It's only used if a user does not pass in a cert pool path during client agent creation.

Reply:
@Baha-sk I had missed that it's used inside that swapCertPool function